assets/app/scripts/services/api.js

```diff
@@ -0,0 +1,255 @@
+'use strict';
+
+// ResourceGroupVersion represents a fully qualified resource
+function ResourceGroupVersion(resource, group, version) {
+  this.resource = resource;
+  this.group    = group;
+  this.version  = version;
+  return this;
+}
+// toString() includes the group and version information if present
+ResourceGroupVersion.prototype.toString = function() {
+  var s = this.resource;
+  if (this.group)   { s += "/" + this.group;   }
+  if (this.version) { s += "/" + this.version; }
+  return s;
+};
+// primaryResource() returns the resource with any subresources removed
+ResourceGroupVersion.prototype.primaryResource = function() {
+  var i = this.resource.indexOf('/');
+  if (i === -1) { return this.resource; }
+  return this.resource.substring(0,i);
+};
+// subresources() returns a (possibly empty) list of subresource segments
+ResourceGroupVersion.prototype.subresources = function() {
+  var i = this.resource.indexOf('/');
```

Comment:
are these two lines needed? if there are no /'s then split would return an array with a single entry and shift would remove that entry, so its still going to return `[]`